packages/app-builder-lib/src/node-module-collector/packageManager.ts

```diff
@@ -0,0 +1,112 @@
+// copy from https://github.com/egoist/detect-package-manager/blob/main/src/index.ts
+// and merge https://github.com/egoist/detect-package-manager/pull/9 to support Monorepo
+import { promises as fs } from "fs"
+import { resolve, dirname } from "path"
+import { exec } from "child_process"
+import { promisify } from "util"
+
+const execa = promisify(exec)
+
+export type PM = "npm" | "yarn" | "pnpm" | "bun"
+
+/**
+ * Check if a path exists
+ */
+async function pathExists(p: string) {
```

Comment:
This function already exists in `builder-util > fs.ts` so we can reuse that from there and avoid duplicate code